### PR TITLE
Shanoir issue#172 Extension request

### PR DESCRIPTION
**Pull request by jcomedouteau**

### CODE REVIEW

File: docker-compose/keycloak/themes/shanoir-theme/login/login.ftl [hunk 1]
```diff
@@ -41,6 +41,9 @@
 									<span><a href="${url.loginResetCredentialsUrl}">${msg("doForgotPassword")}</a></span>
 								</#if>
 							</div>
+							<div>
+							    <a href="${properties.requestExtension!}">Request an extension.</a>
```
Comment (michaelkain) [inline docker-compose/keycloak/themes/shanoir-theme/login/login.ftl:45]: maybe "Account expired?" would better match with the email sent by Shanoir-NG?

File: shanoir-ng-users/src/main/java/org/shanoir/ng/extensionrequest/controller/ExtensionRequestApiController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	@Autowired
+	private KeycloakClient keycloakClient;
+
+	@Autowired
+	private EmailService emailService;
+
+	@Override
+	public ResponseEntity<Void> requestExtension(@RequestBody final ExtensionRequestInfo requestInfo) {
+		try {
+			SecurityContextUtil.initAuthenticationContext("ADMIN_ROLE");
```
Comment (michaelkain) [inline shanoir-ng-users/src/main/java/org/shanoir/ng/extensionrequest/controller/ExtensionRequestApiController.java:47]: Hi Jean-Come, could we do it without that admin_role sec_context? From my point of view, everything down to the service methods, should be open for access. I think it is for security reasons more dangerous to enable an admin_access here then allowing an outsider to ask for an account extension for a given account, especially as the account remains disabled until the extension request is accepted. With kind regards, Michael

Comment (michaelkain) [reply shanoir-ng-users/src/main/java/org/shanoir/ng/extensionrequest/controller/ExtensionRequestApiController.java:47]: open for access for this request

**Review by michaelkain — APPROVED** [review]
in general approved, after code review, please see my minor comments

**Comment by jcomedouteau** [comment]
Done, I created a non protected method to retrieve the user by mail.

**Pull request merged**